api/v1beta1/disk_failure.go

```diff
@@ -25,28 +28,64 @@ type DiskFailureSpec struct {
 	Paths []string `json:"paths"`
 	// +nullable
 	OpenatSyscall *OpenatSyscallSpec `json:"openat,omitempty"`
+	// +kubebuilder:validation:Required
+	// +ddmark:validation:Required=true
+	Probability string `json:"probability"`
 }
 
 // MaxDiskPathCharacters is used to limit the number of characters due to the eBPF memory kernel limitation.
 const MaxDiskPathCharacters = 62
 
 // Validate validates args for the given disruption
-func (s *DiskFailureSpec) Validate() error {
+func (s *DiskFailureSpec) Validate() (retErr error) {
+	if err := s.validatePaths(); err != nil {
+		retErr = multierror.Append(retErr, err)
+	}
+
+	if err := s.validateProbability(); err != nil {
+		retErr = multierror.Append(retErr, err)
+	}
+
+	return
+}
+
+func (s *DiskFailureSpec) validatePaths() error {
 	for _, path := range s.Paths {
-		path := strings.TrimSpace(path)
+		trimSpacePath := strings.TrimSpace(path)
 
-		if path == "" {
+		if trimSpacePath == "" {
 			return fmt.Errorf("the path of the disk failure disruption must not be empty")
 		}
 
-		if len(path) > MaxDiskPathCharacters {
+		if len(trimSpacePath) > MaxDiskPathCharacters {
 			return fmt.Errorf("the path of the disk failure disruption must not exceed %d characters, found %d", MaxDiskPathCharacters, len(path))
 		}
 	}
 
 	return nil
 }
 
+func (s *DiskFailureSpec) validateProbability() error {
+	probabilityError := fmt.Errorf("the probability of the disk failure disruption should be a percentage within the range of 1%% to 100%%")
+
+	if !strings.HasSuffix(s.Probability, "%") {
+		return probabilityError
+	}
+
+	probabilityStr := strings.TrimSuffix(s.Probability, "%")
+	probabilityInt, err := strconv.Atoi(probabilityStr)
+
+	if err != nil {
+		return probabilityError
```

Comment:
It relates to the questioning above, maybe we should not hide to the user it's inputs (or at least what we have as it's input)

It could help understand what does not respect our expectations (or any bug sneaking in the middle if any at some point)
```suggestion
		return fmt.Errorf("Probability percentage can't be converted to int: %w", err)
```